templates/skeleton/app/routes/search.tsx

```diff
@@ -19,9 +19,10 @@ export const meta: MetaFunction = () => {
 export async function loader({request, context}: LoaderFunctionArgs) {
   const url = new URL(request.url);
   const isPredictive = url.searchParams.has('predictive');
-  const searchPromise = isPredictive
-    ? predictiveSearch({request, context})
-    : regularSearch({request, context});
+  const searchPromise: Promise<PredictiveSearchReturn | RegularSearchReturn> =
```

Comment:
For some reason the type of `searchPromise` otherwise is `Promise<PredictiveSearchReturn> | Promise<RegularSearchReturn>`, which meant that `catch` couldn't be invoked on it.